src/cata_algo.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+#ifndef CATA_ALGO_H
+#define CATA_ALGO_H
+
+#include <algorithm>
+#include <functional>
+#include <vector>
+
+namespace algo
+{
+
+/**
+ * @brief Sort elements of a sequence by their rating (the smaller the better).
+ *
+ * @param  begin           An iterator (usual meaning).
+ * @param  end             An iterator (usual meaning).
+ * @param  rating_func     A functor which calculates ratings (the outcome
+ *                         must be comparable using '<' operator).
+ *
+ * The rating is calculated only once per each element.
+ *
+ * @return  Nothing.
```

Comment:
I simply was mimicking the STL way of documenting things. "Since it's a general-purpose function and its signature is similar to `std::sort`'s, why wouldn't I document it similarly?" I thought. BTW `std::sort`'s documentation is a bit funny:
```c++
   *  @param  __first   An iterator.
   *  @param  __last    Another iterator.
```